File: php-advisor-core/src/Payloads/RadiusPayload.php
<?php

namespace StormGeo\AdvisorCore\Payloads;

class RadiusPayload extends BasePayload
{
  /**
   * @var int
   */
	public $localeId;

  /**
   * @var string
   */
	public $latitude;

  /**
   * @var string
   */
	public $longitude;

  /**
   * @var string
   */
	public $startDate;

  /**
   * @var string
   */
  public $endDate;

  /**
   * @var int
   */
  public $radius;

  /**
   * @param array{localeId:int,latitude:string,longitude:string,startDate:string,endDate:string,radius:int} $parameters
   */
  public function __construct($parameters = [])
  {
    parent::__construct(
      ['localeId', 'latitude', 'longitude', 'startDate', 'endDate', 'radius'],
      $parameters
    );
  }

  /**
   * @var array
   */
  public function getQueryParams()
  {
    return (array) $this;
  }
}

File: dotnet-advisor-core/Payloads/RadiusPayload.cs
namespace StormGeo.AdvisorCore.Payloads;

public class RadiusPayload
{
    public int? LocaleId { get; set; }
    public string? Latitude { get; set; }
    public string? Longitude { get; set; }
    public string? StartDate { get; set; }
    public string? EndDate { get; set; }
    public int? Radius { get; set; }

    public string GetQueryParams()
    {
        return new QueryParameterBuilder()
            .AddLocaleId(LocaleId)
            .AddLatLon(Latitude, Longitude)
            .AddStartDate(StartDate)
            .AddEndDate(EndDate)
            .AddRadius(Radius)
            .Build();
    }
}
